## 1. Importing the necessary packages

In [ ]:
# Importing the necessary function
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score,classification_report,confusion_matrix

In [ ]:
#Importing & defining the datframe
df=pd.read_csv('dataset_35_dermatology.csv')
df.head()

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  \
0                           1                  0  ...   
1                           1                  1  ...   
2                           0                  0  ...   
3                           3                  2  ...   
4                           0                  0  ...   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate  Age  class  
0                     0   55      2  
1                     0    8      1  
2                     3   26      3  
3                     0   40      1  
4                     3   45      3  

[5 rows x 35 columns]

## 2. Basic checks

In [ ]:
df.shape # DataFrame shape

(366, 35)

In [ ]:
df.size # DataFrame size

12810

In [ ]:
df.info() # Checking the information

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 366 entries, 0 to 365
Data columns (total 35 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   erythema                                  366 non-null    int64  
 1   scaling                                   366 non-null    int64  
 2   definite_borders                          366 non-null    int64  
 3   itching                                   366 non-null    int64  
 4   koebner_phenomenon                        366 non-null    int64  
 5   polygonal_papules                         366 non-null    int64  
 6   follicular_papules                        366 non-null    int64  
 7   oral_mucosal_involvement                  366 non-null    int64  
 8   knee_and_elbow_involvement                366 non-null    int64  
 9   scalp_involvement                         366 non-null    int64  
 10  family_history                        

## 3. Checking the null Values

In [ ]:
#Replacing the special charecters
df.replace('?',np.nan,inplace=True)
df.head(5)

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  \
0                           1                  0  ...   
1                           1                  1  ...   
2                           0                  0  ...   
3                           3                  2  ...   
4                           0                  0  ...   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate  Age  class  
0                     0   55      2  
1                     0    8      1  
2                     3   26      3  
3                     0   40      1  
4                     3   45      3  

[5 rows x 35 columns]

## 4 Replacing The nulll values with mean

In [ ]:
# Filling the null values with mean
columns = ['Age']
for col in columns:
    df[col]=pd.to_numeric(df[col])
    df[col].fillna(df[col].mean(), inplace=True)
df.head() 

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  \
0                           1                  0  ...   
1                           1                  1  ...   
2                           0                  0  ...   
3                           3                  2  ...   
4                           0                  0  ...   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate   Age  class  
0                     0  55.0      2  
1                     0   8.0      1  
2                     3  26.0      3  
3                     0  40.0      1  
4                     3  45.0      3  

[5 rows x 35 columns]

In [ ]:
df.isnull().sum()

erythema                                    0
scaling                                     0
definite_borders                            0
itching                                     0
koebner_phenomenon                          0
polygonal_papules                           0
follicular_papules                          0
oral_mucosal_involvement                    0
knee_and_elbow_involvement                  0
scalp_involvement                           0
family_history                              0
melanin_incontinence                        0
eosinophils_in_the_infiltrate               0
PNL_infiltrate                              0
fibrosis_of_the_papillary_dermis            0
exocytosis                                  0
acanthosis                                  0
hyperkeratosis                              0
parakeratosis                               0
clubbing_of_the_rete_ridges                 0
elongation_of_the_rete_ridges               0
thinning_of_the_suprapapillary_epi

In [ ]:
df.shape # Shape of a DataFrame

(366, 35)

## 5. Defining the 'X' and 'y'

In [ ]:
X = df.iloc[:,:34]
y=df['class']

In [ ]:
X.head(5)

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
0         2        2                 0        3                   0   
1         3        3                 3        2                   1   
2         2        1                 2        3                   1   
3         2        2                 2        0                   0   
4         2        3                 2        2                   2   

   polygonal_papules  follicular_papules  oral_mucosal_involvement  \
0                  0                   0                         0   
1                  0                   0                         0   
2                  3                   0                         3   
3                  0                   0                         0   
4                  2                   0                         2   

   knee_and_elbow_involvement  scalp_involvement  ...  focal_hypergranulosis  \
0                           1                  0  ...                      0   
1                           1                  1  ...                      0   
2                           0                  0  ...                      2   
3                           3                  2  ...                      0   
4                           0                  0  ...                      2   

   disappearance_of_the_granular_layer  \
0                                    0   
1                                    0   
2                                    0   
3                                    3   
4                                    2   

   vacuolisation_and_damage_of_basal_layer  spongiosis  \
0                                        0           3   
1                                        0           0   
2                                        2           3   
3                                        0           0   
4                                        3           2   

   saw-tooth_appearance_of_retes  follicular_horn_plug  \
0                              0                     0   
1                              0                     0   
2                              2                     0   
3                              0                     0   
4                              3                     0   

   perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
0                             0                                    1   
1                             0                                    1   
2                             0                                    2   
3                             0                                    3   
4                             0                                    2   

   band-like_infiltrate   Age  
0                     0  55.0  
1                     0   8.0  
2                     3  26.0  
3                     0  40.0  
4                     3  45.0  

[5 rows x 34 columns]

In [ ]:
y.head()

0    2
1    1
2    3
3    1
4    3
Name: class, dtype: int64

## 6. Model-2=> Support Vector Machine

In [ ]:
from sklearn.svm import SVC #Importing the model and train_test_split packages
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score,classification_report,confusion_matrix # importing packages for model evaluation

In [ ]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=25)

In [ ]:
X_train.head()

erythema  scaling  definite_borders  itching  koebner_phenomenon  \
173         3        3                 3        0                   0   
148         2        2                 2        0                   2   
176         3        2                 3        0                   0   
153         2        2                 2        0                   0   
354         2        0                 1        2                   0   

     polygonal_papules  follicular_papules  oral_mucosal_involvement  \
173                  0                   0                         0   
148                  0                   0                         0   
176                  0                   0                         0   
153                  0                   0                         0   
354                  0                   0                         0   

     knee_and_elbow_involvement  scalp_involvement  ...  \
173                           2                  0  ...   
148                           0                  0  ...   
176                           2                  2  ...   
153                           1                  0  ...   
354                           0                  0  ...   

     focal_hypergranulosis  disappearance_of_the_granular_layer  \
173                      0                                    2   
148                      0                                    0   
176                      0                                    2   
153                      0                                    0   
354                      0                                    0   

     vacuolisation_and_damage_of_basal_layer  spongiosis  \
173                                        0           0   
148                                        0           2   
176                                        0           0   
153                                        0           0   
354                                        0           0   

     saw-tooth_appearance_of_retes  follicular_horn_plug  \
173                              0                     0   
148                              0                     0   
176                              0                     0   
153                              0                     0   
354                              0                     0   

     perifollicular_parakeratosis  inflammatory_monoluclear_inflitrate  \
173                             0                                    2   
148                             0                                    1   
176                             0                                    2   
153                             0                                    1   
354                             0                                    2   

     band-like_infiltrate   Age  
173                     0  18.0  
148                     0  30.0  
176                     0  33.0  
153                     0  52.0  
354                     0  56.0  

[5 rows x 34 columns]

In [ ]:
y_train.head(5)

173    1
148    4
176    1
153    1
354    5
Name: class, dtype: int64

### 6.2 Training and defining the model

In [ ]:
#Defining the model
model= SVC()

In [ ]:
model.fit(X_train,y_train)# fitting the data
y_predict= model.predict(X_test)

### 6.3 Model Evaluation

In [ ]:
accuracy_score(y_test,Y_predict)

0.6621621621621622

In [ ]:
print(classification_report(y_test,y_predict)) # Classification report

              precision    recall  f1-score   support

           1       0.70      1.00      0.82        21
           2       0.55      0.33      0.41        18
           3       1.00      1.00      1.00        15
           4       0.18      0.25      0.21         8
           5       1.00      0.22      0.36         9
           6       0.60      1.00      0.75         3

    accuracy                           0.66        74
   macro avg       0.67      0.63      0.59        74
weighted avg       0.70      0.66      0.63        74



In [ ]:
confusion_matrix(y_test,y_predict) # Confusion matrix

array([[21,  0,  0,  0,  0,  0],
       [ 2,  6,  0,  9,  0,  1],
       [ 0,  0, 15,  0,  0,  0],
       [ 1,  4,  0,  2,  0,  1],
       [ 6,  1,  0,  0,  2,  0],
       [ 0,  0,  0,  0,  0,  3]], dtype=int64)

### 6.4 Model Evaluation with Cross validation

In [ ]:
from sklearn.model_selection import cross_val_score
scores = cross_val_score(model,X,y,cv=10,scoring="accuracy")
print(scores)

[0.64864865 0.72972973 0.7027027  0.72972973 0.78378378 0.72972973
 0.75       0.77777778 0.75       0.69444444]


In [ ]:
print("Average Accuracy:",scores.mean())
print("Std:",scores.std())
#std of < 0.05 is good.

Average Accuracy: 0.7296546546546547
Std: 0.038212328231550116


## The Avarage accuracy Score of Support Vector Machine model with Cross_val is = 0.72965

Please prefer other notebook for different model